mycore-solr/src/main/java/org/mycore/solr/common/xml/MCRSolrQueryResolver.java

```diff
@@ -75,77 +68,38 @@ public class MCRSolrQueryResolver implements URIResolver {
-            Optional<String> requestHandler = Optional.ofNullable(matcher.group(REQUEST_HANDLER_GROUP_NAME));
-            Optional<String> query = Optional.ofNullable(matcher.group(QUERY_GROUP_NAME));
-
-            if (!newMatcher.matches()) {
-                printMismatchWarning(href);
-            } else {
-                String newCore = newMatcher.group(CORE_GROUP_NAME);
-                String newRequestHandler = newMatcher.group(REQUEST_HANDLER_GROUP_NAME);
-                String newQuery = newMatcher.group(QUERY_GROUP_NAME);
-                if (!Objects.equals(core.orElse(null), newCore) ||
-                    !Objects.equals(requestHandler.orElse(null), newRequestHandler) ||
-                    !Objects.equals(query.orElse(null), newQuery)) {
-                    printMismatchWarning(href);
-                }
-            }
-
-            HttpSolrClientBase client = core.flatMap(MCRSolrCoreManager::get)
-                .map(MCRSolrCore::getClient)
-                .orElse((HttpSolrClientBase) MCRSolrCoreManager.getMainSolrClient());
-
-            if (query.isPresent()) {
-                MCRSolrURL solrURL = new MCRSolrURL(client, query.get());
-                requestHandler.map("/"::concat).ifPresent(solrURL::setRequestHandler);
-
-                URI solrRequestURI;
-                try {
-                    solrRequestURI = solrURL.getUrl().toURI();
-                } catch (URISyntaxException e) {
-                    throw new IllegalArgumentException("Could not create URI from " + solrURL.getUrl(), e);
-                }
-
-                HttpRequest.Builder solrRequestBuilder = MCRSolrUtils.getRequestBuilder().uri(solrRequestURI);
-
-                MCRSolrAuthenticationManager.obtainInstance()
-                    .applyAuthentication(solrRequestBuilder, MCRSolrAuthenticationLevel.SEARCH);
-
-                try (HttpClient httpClient = MCRHttpUtils.getHttpClient()) {
-                    HttpRequest solrRequest = solrRequestBuilder.build();
-                    HttpResponse<byte[]> response = httpClient.send(solrRequest,
-                        HttpResponse.BodyHandlers.ofByteArray());
-
-                    if (response.statusCode() != 200) {
-                        throw new MCRException(
-                            "Error while executing request: " + response.version() + " " + response.statusCode());
-                    }
-
-                    MCRByteContent result = new MCRByteContent(response.body());
-                    result.setSystemId(solrRequest.uri().toString());
-                    return result.getSource();
-                } catch (InterruptedException | IOException e) {
-                    throw new MCRException("Error while executing request", e);
-                }
-            }
+        Matcher matcher = URI_PATTERN.matcher(href);
+
+        if (!matcher.matches()) {
+            printMismatchWarning(href);
```

Comment:
Nitpick: the mismatch warning should may point to the javadoc of the class. The Ticket MCR-2872 does not really give a clue what is wrong.